barretenberg/cpp/src/barretenberg/flavor/goblin_ultra_recursive.hpp

```diff
@@ -104,6 +106,12 @@ template <typename BuilderType> class GoblinUltraRecursive_ {
      */
     class VerificationKey : public VerificationKey_<GoblinUltra::PrecomputedEntities<Commitment>> {
       public:
+        VerificationKey(const size_t circuit_size, const size_t num_public_inputs)
+        {
+            this->circuit_size = circuit_size;
+            this->log_circuit_size = numeric::get_msb(circuit_size);
+            this->num_public_inputs = num_public_inputs;
```

Comment:
So what's happening in the vanilla ultra recursive verifier is that the verification key is created from the native one, the method below.
 On the other hand, this constructor already existed in hte Ultra and GoblinUltra native flavors. The decider, at least in the current state, receives the folded verification key as part of the proof as well as the `circuit_size` and `public_input_size`. It needs a constructor to create a vk object so that it can then iterate over the rest of its fields (the selector commitments) and assign them whatever it received in the proof. Open to suggestions if you think there's a better way to do this, I thought that following what's in the native one is best to minimise the tiny differences.